vignettes/articles/writing-custom-fns.Rmd

```diff
@@ -0,0 +1,265 @@
+
+</div>
+
+In addition to these, custom check functions can also have additional arguments for inputs required by the check. Some of these inputs **are available in the check caller environment** and can be passed automatically to a custom check function by including an argument with the same name as the input object required in the custom function formals. Other inputs can be passed explicitly to function arguments through [the functions `args` field](deploying-custom-functions#validations-yml-structure) when configuring the `validations.yml` file.
+
+### Arguments available in the caller environment 
+
+```{r child="children/_custom-fn-available-args.Rmd", echo=FALSE, results="asis"}
+```
+
+<div class="alert alert-info">
+
+Note that, when writing custom functions, these objects **do not all need to be specified as arguments** in the function definition. **Only the ones that your custom function actually requires as inputs**.
+
+</div>
+
+### Additional arguments
+
+You can add additional arguments to custom check functions and pass values to them by including them in the `args` configuration in the `validations.yml` file. These values are passed to the custom check function by `hubValidations` when the function is called.
+
+
+If you do add additional arguments to a custom check function, you should also add input  checks at the start of the function to ensure inputs are valid. [The `checkmate` package](https://mllg.github.io/checkmate/) contains a wide range of functions for checking inputs.
+
+For example, the optional check `opt_check_tbl_col_timediff()` (which is deployed in exactly the same fashion as custom functions, i.e. through the `validations.yml` file) takes additional arguments `t0_colname`, `t1_colname` and `timediff`. 
+
+
+```{r, comment = NA}
+opt_check_tbl_col_timediff
+```
+
+
+You can add an example extra argument with `extra_args = TRUE` when creating the custom check function with `create_custom_check()`.
+
+```{r}
+create_custom_check("cstm_check_tbl_args",
+  hub_path = hub_path,
+  extra_args = TRUE
+)
+```
+
+This adds an extra example argument `extra_arg` to the custom check function formals as well as an example input check to the top of the function body.
+
+```{r, code=readLines(r_path(hub_path, "cstm_check_tbl_args")), eval=FALSE, comment=NA}
+```
+
+
+## Function output
+
+### Capturing and returning check results
+
+**Returning a check condition with `capture_check_cnd()`**
```

Comment:
This looks like it's meant to be a heading, so it should be coded as such. 

```suggestion
### Returning a check condition with `capture_check_cnd()`
```

Reply:
It's more of a subtitle really than a heading